### PR TITLE
fix: [CDS-77432]: Improve Terragrunt identifier log message

### PR DESCRIPTION
**Pull request by sarthakkasat**
## Describe your changes
Provision Identifier cannot contain hyphen

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51900)

### CODE REVIEW

**Comment by vlica-harness** [comment]
Hi @sarthakkasat 
is this PR only for this message improvement, or also for another issue mentioned in the ticket?

**Comment by sarthakkasat** [comment]
This is just for error message improvement, will create a separate PR for other issue

File: 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terragrunt/TerragruntStepHelper.java [hunk 1]
```diff
@@ -246,8 +246,9 @@ public String generateFullIdentifier(String provisionerIdentifier, Ambiance ambi
       return format("%s/%s/%s/%s", AmbianceUtils.getAccountId(ambiance), AmbianceUtils.getOrgIdentifier(ambiance),
           AmbianceUtils.getProjectIdentifier(ambiance), provisionerIdentifier);
     } else {
-      throw new InvalidRequestException(
-          format("Provisioner Identifier cannot contain special characters or spaces: [%s]", provisionerIdentifier));
+      throw new InvalidRequestException(format(
+          "Provisioner Identifier must start with a letter or _ and can then be followed by alphanumerics or _: [%s]",
+          provisionerIdentifier));
```
Comment (vlica-harness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terragrunt/TerragruntStepHelper.java:251]: what do you think  if we add the previous message as well, saying about those restricted "special characters, spaces, hyphens" ?

**Review by yogesh-chauhan — APPROVED** [review]

File: 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terragrunt/TerragruntStepHelper.java (hunk 1 above)
Comment (vlica-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terragrunt/TerragruntStepHelper.java:251]: ok, I think anyway we will have questions in future even saying how the identifier should be :D

**Review by vlica-harness — APPROVED** [review]

**Comment by sarthakkasat** [comment]
trigger uts

**Comment by sarthakkasat** [comment]
trigger messagecheck

**Pull request merged**